### PR TITLE
docs(vars): Add some documentation explaining the usage and limits of vars

### PR DESCRIPTION
**Pull request by Rio**
Currently it's only in the example `kustomization.yaml` but I imagine adding it to the readme would also be nice.

### CODE REVIEW

File: docs/kustomization.yaml [hunk 1]
```diff
@@ -136,3 +136,43 @@ patches:
...
+      apiVersion: v1
+    fieldref:
+      fieldpath: metadata.name
+  - name: ANOTHER_DEPLOYMENTS_POD_RESTART_POLICY
+    objref:
+      kind: Deployment
+      name: my-deployment
+      apiVersion: apps/v1
+    fieldref:
+      fieldpath: spec.template.spec.restartPolicy
```
Comment (Liujingfang1) [inline docs/kustomization.yaml]: new line

File: docs/kustomization.yaml [hunk 2]
```diff
@@ -136,3 +136,43 @@ patches:
 crds:
 - crds/typeA.yaml
 - crds/typeB.yaml
+
+# Vars are used to insert values from resources that cannot be referenced
+# otherwise. For example if you need to pass a Service's name to the arguments
+# or environment variables of a program but without hard coding the actual name
+# of the Service you'd insert `$(MY_SERVICE_NAME)` into the value field of the
```
Comment (Liujingfang1) [inline docs/kustomization.yaml]: Can you add  a few lines showing how to insert `$(MY_SERVICE_NAME)` into container args or env

**Comment by Rio** [comment]
@Liujingfang1, added your suggestions.

**Comment by Liujingfang1** [comment]
Looks good. Please squash the 5 commits into 1.

**Comment by Rio** [comment]
Squashed

**Pull request merged**